gsoc_contributors_guide.md

```diff
@@ -5,10 +5,104 @@ The Echoshader team aims to recruit talented [Google Summer of Code (GSoC)](http
 
 ## Getting Started
 
```

Comment:
I have a general suggestion here that it would be good to limit the line length so that the text is easier to read, especially on a small screen the wrap-around can be disruptive. See the recommendation in [black](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#line-length). For echopype we have it [configured to 100](https://github.com/OSOceanAcoustics/echopype/blob/5cf1153b3d292ce819276ed12c86816c694b75ec/setup.cfg#L21-L25) but I would recommend maybe 90.

Reply:
@leewujung how exactly do I enable the limiting of the line length on Github for a markdown file? The links you provided seem to be Python specific.

Reply:
I pointed to those just as references for why it's good to do so, but I was thinking to just break it manually here would be fine, since in this repo the main work will still be code and not text.

But, I did a quick google search when seeing your question. I think we can use tools such as [markdownlint](`https://github.com/markdownlint/markdownlint`) as part of pre-commit so that this is automated also. 

I'll make an issue for creating pre-commit and config for these. We don't have code here just yet for black or flake8 to lint, but maybe can set up markdownlint to get the mechanism in place. Interested? ;)

Reply:
I've never set these up on my own, but I can give it a try. Go ahead and assign me to the issue you create.